watertap/tools/oli_api/flash.py

```diff
@@ -75,420 +73,1163 @@
+        :param use_scaling_rigorous: bool to switch between Rigorous (default) and Estimated scaling computations
+        :param file_name: string for file to write, if any
+
+        :return json_input: JSON for Water Analysis
         """
 
-        self.water_analysis_input_list = []
+        _logger.info(f"Configuring {flash_method} Flash JSON ...")
+
+        if flash_method not in [
+            "isothermal",
+            "isenthalpic",
+            "bubblepoint",
+            "dewpoint",
+            "vapor-amount",
+            "vapor-fraction",
+            "isochoric",
+            "setph",
+            "precipitation-point",
+            "corrosion-rates",
+        ]:
+            raise RuntimeError(
+                f"Failed to configure Flash. Invalid method: {flash_method}"
+            )
 
-        # build entries for temperature and pressure
-        self.water_analysis_properties["Temperature"].update(
-            {"value": state_vars["temperature"]}
-        )
-        self.water_analysis_input_list.append(
-            self.water_analysis_properties["Temperature"]
-        )
-        self.water_analysis_properties["Pressure"].update(
-            {"value": state_vars["pressure"]}
-        )
-        self.water_analysis_input_list.append(
-            self.water_analysis_properties["Pressure"]
-        )
+        input_dict = {}
 
-        for component in state_vars["components"]:
-            charge = get_charge(component)
-            name = get_oli_name(component)
-            conc = self._oli_units(
-                state_vars["components"][component], state_vars["units"]["components"]
-            )
+        if not inflows:
+            raise RuntimeError("Inflows must be defined for Flash Analysis.")
+
+        temp_input = {
+            "unit": self.input_unit_set["temperature"]["oli_unit"],
+            "value": 273.15,
+        }
+        if temperature:
+            if float(temperature):
+                temp_input.update({"value": float(temperature)})
+            else:
+                raise RuntimeError(
+                    f"Invalid temperature: {temperature}. Expected number"
+                )
+        input_dict["temperature"] = temp_input
+
+        pres_input = {
+            "unit": self.input_unit_set["pressure"]["oli_unit"],
+            "value": 101325,
+        }
+        if pressure:
+            if float(pressure):
+                pres_input.update({"value": float(pressure)})
+            else:
+                raise RuntimeError(f"Invalid pressure: {pressure}. Expected number")
+        input_dict["pressure"] = pres_input
+
+        if flash_method in [
+            "bubblepoint",
+            "dewpoint",
+            "vapor-amount",
+            "vapor-fraction",
+            "isochoric",
+        ]:
+            if calculated_variable:
+                _logger.info(
+                    f"{flash_method} will calculate {calculated_variable} as its variable"
```

Comment:
As mentioned in previous comment, I think this is where you want to add some check(s) to ensure that the specified calculated variable is valid. Also, what if I want to enter more than one calculated variable? A test to verify this functionality would be good.